ruby_programming/advanced_ruby/pattern_matching.md

```diff
@@ -652,3 +652,12 @@ Pattern Matching provides a powerful way to get at the data you need in a way th
 1. Start with the [Ruby docs](https://docs.ruby-lang.org/en/3.0.0/doc/syntax/pattern_matching_rdoc.html) on pattern matching. They cover quite a lot of ground.
 2. One thing we didn't cover here is matching your own objects. We did mention implementing the `===` method but there are a couple of other methods you can implement that will allow you to use array and hash pattern matching. Check out [this guide to pattern matching](https://rubyreferences.github.io/rubyref/language/pattern-matching.html), particularly [this section](https://rubyreferences.github.io/rubyref/language/pattern-matching.html#matching-non-primitive-objects-deconstruct_keys-and-deconstruct).
 </div>
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href="https://docs.ruby-lang.org/en/3.0.0/doc/syntax/pattern_matching_rdoc.html">What is pattern matching?</a>
+- <a class="knowledge-check-link" href="https://docs.ruby-lang.org/en/3.0.0/doc/syntax/pattern_matching_rdoc.html#label-Patterns">What is the difference between array patterns and hash patterns?</a> 
+- <a class="knowledge-check-link" href="#variable-pattern">What is the variable pattern match?</a>
+- <a class="knowledge-check-link" href="#array-pattern-match">Which different ways of array pattern matching are there?</a>
```

Comment:
I think this would read better as `What are the different ways of pattern matching on an array?`